### PR TITLE
feat(config): add support for HomeSeer HS-FLS100-G2

### PR DESCRIPTION
**Pull request by ibddot**
Adds support for HomeSeer HS-FLS100-G2 floodlight sensor.

Adds temperature and few more config parameters compared to previous device version (HS-FLS100).
Data from product manual and Z-Wave Alliance.

### CODE REVIEW

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"supportsZWavePlus": true,
+	"paramInformation": {
+		"1": {
+			"label": "PIR trigger off period",
+			"description": "Period to send Trigger Off command after PIR is triggered.",
+			"unit": "Seconds",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"unit": "seconds",
```

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			"valueSize": 2,
+			"minValue": 8,
+			"maxValue": 720,
+			"defaultValue": 180,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"2": {
+			"label": "Lux sensor threshold",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"label": "Illuminance sensor threshold",
```
?

Comment (AlCalzone) [reply packages/config/config/devices/0x000c/hs-fls100-g2.json]: Should also have `"unit": "lux"` I guess

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+			"minValue": 10,
+			"maxValue": 900,
+			"defaultValue": 50,
+			"readOnly": false,
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"3": {
+			"label": "Auto Report luminance and temperature interval",
+			"description": "set the interval of periodic lux level report and temperature report to the controller",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]: remove description, it does not add value

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+			"unit": "min",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"unit": "minutes",
```
for consistency with other files

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+			"options": [
+				{
+					"label": "disable alert",
+					"value": 0
+				},
+				{
+					"label": "enable alert",
+					"value": 1
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"options": [
				{
					"label": "Disable alert",
					"value": 0
				},
				{
					"label": "Enable alert",
					"value": 1
				}
			]
```

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+			"options": [
+				{
+					"label": "controlled by Z wave controller only",
+					"value": 0
+				},
+				{
+					"label": "controlled by PIR trigger and Lux level, or by Z wave controller",
+					"value": 1
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"options": [
				{
					"label": "controlled by Z-Wave controller only",
					"value": 0
				},
				{
					"label": "controlled by PIR trigger and Lux level, or by Z-Wave controller",
					"value": 1
				}
			]
```

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+			"options": [
+				{
+					"label": "floodlight is controlled by Lux level and PIR trigger",
+					"value": 0
+				},
+				{
+					"label": "floodlight is controlled by Lux level only, regardless of PIR trigger",
+					"value": 1
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"options": [
				{
					"label": "Floodlight is controlled by Lux level and PIR trigger",
					"value": 0
				},
				{
					"label": "Floodlight is controlled by Lux level only, regardless of PIR trigger",
					"value": 1
				}
			]
```

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+					"value": 0
+				},
+				{
+					"label": "controlled by PIR trigger and Lux level, or by Z wave controller",
+					"value": 1
+				}
+			]
+		},
+		"6": {
+			"label": "Lux sensor",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"label": "Floodlight triggers",
```
maybe? Lux sensor does not make sense in this context

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+					"value": 0
+				},
+				{
+					"label": "floodlight is controlled by Lux level only, regardless of PIR trigger",
+					"value": 1
+				}
+			]
+		},
+		"7": {
+			"label": "Mesured temperature offset",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"label": "Measured temperature offset",
```

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 127,
+			"defaultValue": 0,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]: Which unit? Does this only allow positive offsets?

File: packages/config/config/devices/0x000c/hs-fls100-g2.json [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+				},
+				{
+					"label": "mid level, approx. 10m distance",
+					"value": 1
+				},
+				{
+					"label": "high level, approx. 20m distance",
+					"value": 2
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x000c/hs-fls100-g2.json]:
```suggestion
			"options": [
				{
					"label": "Low level, approx. 6m distance",
					"value": 0
				},
				{
					"label": "Mid level, approx. 10m distance",
					"value": 1
				},
				{
					"label": "High level, approx. 20m distance",
					"value": 2
				}
			]
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: packages/config/config/devices/0x000c/hs-fls100-g2.json (hunk 8 above)
Comment (ibddot) [reply packages/config/config/devices/0x000c/hs-fls100-g2.json]: Definitely agree that makes more sense.

File: packages/config/config/devices/0x000c/hs-fls100-g2.json (hunk 10 above)
Comment (ibddot) [reply packages/config/config/devices/0x000c/hs-fls100-g2.json]: After looking at Z-wave Alliance description closer it looks like it should be -100 to 100 with units of 0.1 °C.


```json
    <ConfigurationParameterExport>
      <Name>Temperature Sensor Offset</Name>
      <Description>0x9C - 0x64 (offset range : -10.0°C ~ +10.0°C)</Description>
      <ParameterNumber>7</ParameterNumber>
      <Size>1</Size>
      <DefaultValue>0</DefaultValue>
      <pType>1</pType>
      <flagReadOnly>false</flagReadOnly>
      <flagReInclude>false</flagReInclude>
      <flagAdvanced>false</flagAdvanced>
      <v3plus>false</v3plus>
      <minValue>0</minValue>
      <maxValue>127</maxValue>
      <ConfigurationParameterValues>
        <ConfigurationParameterValueExport>
          <From>0</From>
          <To>127</To>
          <Description />
        </ConfigurationParameterValueExport>
      </ConfigurationParameterValues>
    </ConfigurationParameterExport>
```

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**